### PR TITLE
update type field and move loop logic to controller

### PR DESCRIPTION
**Pull request by justinpjose**
### What
**Describe what you have changed and why.**
1. Change the field names in `ContentType` struct from `Type` -> `Group` and `SubTypes` -> `Types`. This gives clarity on what each field means. This is an example of how each field links to each other:

```
Publication (Category)
-> Article (Group)
-> -> "articles","static_articles" (Types)
```

2. Moved the logic to get the search result `Type` `LocaliseKey` from the template to the controller 

### How to review
**Describe the steps required to test the changes.**
- Check if the code changes makes sense
- Run locally and see if the search page is functioning as normal
- Check if the tests pass

### Who can review
**Describe who worked on the changes, so that other people can review.**
Anyone

### CODE REVIEW

**Review by NoelleDL — DISMISSED** [review]

File: data/filter.go [hunk 1]
```diff
@@ -72,75 +72,76 @@ var (
...
+		Group:           "bulletin",
+		Types:           []string{"bulletin"},
 	}
 
 	// Article - Search information specific for articles
 	Article = ContentType{
 		LocaliseKeyName: "Article",
-		Type:            "article",
-		SubTypes:        []string{"article", "article_download"},
+		Group:           "article",                               //Type
```
Comment (nshumoogum) [inline data/filter.go:82]: Probably need to remove comments on this line and the one below

File: data/filter.go [hunk 2]
```diff
@@ -72,75 +72,76 @@ var (
...
+		Types:           []string{"timeseries"},
 	}
 
 	// Datasets - Search information specific for datasets
 	Datasets = ContentType{
 		LocaliseKeyName: "Datasets",
-		Type:            "datasets",
-		SubTypes:        []string{"dataset_landing_page", "timeseries_dataset"},
+		Group:           "datasets",
+		Types:           []string{"dataset_landing_page", "reference_tables"},
```
Comment (nshumoogum) [inline data/filter.go:104]: Im not sure why this has changed back to `reference_tables` from `timeseries_dataset`

File: handlers/handlers.go [hunk 3]
```diff
@@ -134,7 +134,7 @@ func setCountToCategories(ctx context.Context, countResp searchCli.Response, cat
 	categoryLoop:
 		for i, category := range categories {
 			for j, contentType := range category.ContentTypes {
-				for _, subType := range contentType.SubTypes {
+				for _, subType := range contentType.Types {
```
Comment (nshumoogum) [inline handlers/handlers.go:137]: Probably should remove empty file `helll` ;)

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: data/filter.go (hunk 1 above)
Comment (justinpjose) [reply data/filter.go]: Done

File: data/filter.go (hunk 2 above)
Comment (justinpjose) [reply data/filter.go]: No idea either but this has been updated

File: handlers/handlers.go (hunk 3 above)
Comment (justinpjose) [reply handlers/handlers.go:137]: Sorry about that

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**